### PR TITLE
get valid VM size types for azure provider

### LINKED ISSUE #2511: Display only valid values in field "Node size" (Azure)
As an end user, I want only have a list of valid Node sizes for my chosen, datacenter displayed in the UI.
image

Like in this example for Azure datacenter region `westeurope` I get shown up nodes, what are not valid. In combination with #903 this is really painful. 

For Azure there is a way to ask which machine type is valid for the a region:
```
$ az vm list-skus --location westeurope --size Standard_DS --output table 

ResourceType     Locations    Name                Zones    Capabilities                                                                                                                                                                            Restrictions
---------------  -----------  ------------------  -------  --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------  --------------
virtualMachines  westeurope   Standard_DS1_v2     1,2,3    ['MaxResourceVolumeMB=7168', 'OSVhdSizeMB=1047552', 'vCPUs=1', 'MemoryGB=3.5', 'MaxDataDiskCount=4', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']      None
virtualMachines  westeurope   Standard_DS2_v2     1,2,3    ['MaxResourceVolumeMB=14336', 'OSVhdSizeMB=1047552', 'vCPUs=2', 'MemoryGB=7', 'MaxDataDiskCount=8', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']       None
virtualMachines  westeurope   Standard_DS3_v2     1,2,3    ['MaxResourceVolumeMB=28672', 'OSVhdSizeMB=1047552', 'vCPUs=4', 'MemoryGB=14', 'MaxDataDiskCount=16', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']     None
virtualMachines  westeurope   Standard_DS4_v2     1,2,3    ['MaxResourceVolumeMB=57344', 'OSVhdSizeMB=1047552', 'vCPUs=8', 'MemoryGB=28', 'MaxDataDiskCount=32', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']     None
virtualMachines  westeurope   Standard_DS5_v2     1,2,3    ['MaxResourceVolumeMB=114688', 'OSVhdSizeMB=1047552', 'vCPUs=16', 'MemoryGB=56', 'MaxDataDiskCount=64', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']   None
virtualMachines  westeurope   Standard_DS11-1_v2  1,2,3    ['MaxResourceVolumeMB=28672', 'OSVhdSizeMB=1047552', 'vCPUs=2', 'MemoryGB=14', 'MaxDataDiskCount=8', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']      None
virtualMachines  westeurope   Standard_DS11_v2    1,2,3    ['MaxResourceVolumeMB=28672', 'OSVhdSizeMB=1047552', 'vCPUs=2', 'MemoryGB=14', 'MaxDataDiskCount=8', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']      None
virtualMachines  westeurope   Standard_DS12-1_v2  1,2,3    ['MaxResourceVolumeMB=57344', 'OSVhdSizeMB=1047552', 'vCPUs=4', 'MemoryGB=28', 'MaxDataDiskCount=16', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']     None
virtualMachines  westeurope   Standard_DS12-2_v2  1,2,3    ['MaxResourceVolumeMB=57344', 'OSVhdSizeMB=1047552', 'vCPUs=4', 'MemoryGB=28', 'MaxDataDiskCount=16', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']     None
virtualMachines  westeurope   Standard_DS12_v2    1,2,3    ['MaxResourceVolumeMB=57344', 'OSVhdSizeMB=1047552', 'vCPUs=4', 'MemoryGB=28', 'MaxDataDiskCount=16', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']     None
virtualMachines  westeurope   Standard_DS13-2_v2  1,2,3    ['MaxResourceVolumeMB=114688', 'OSVhdSizeMB=1047552', 'vCPUs=8', 'MemoryGB=56', 'MaxDataDiskCount=32', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']    None
virtualMachines  westeurope   Standard_DS13-4_v2  1,2,3    ['MaxResourceVolumeMB=114688', 'OSVhdSizeMB=1047552', 'vCPUs=8', 'MemoryGB=56', 'MaxDataDiskCount=32', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']    None
virtualMachines  westeurope   Standard_DS13_v2    1,2,3    ['MaxResourceVolumeMB=114688', 'OSVhdSizeMB=1047552', 'vCPUs=8', 'MemoryGB=56', 'MaxDataDiskCount=32', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']    None
virtualMachines  westeurope   Standard_DS14-4_v2  1,2,3    ['MaxResourceVolumeMB=229376', 'OSVhdSizeMB=1047552', 'vCPUs=16', 'MemoryGB=112', 'MaxDataDiskCount=64', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']  None
virtualMachines  westeurope   Standard_DS14-8_v2  1,2,3    ['MaxResourceVolumeMB=229376', 'OSVhdSizeMB=1047552', 'vCPUs=16', 'MemoryGB=112', 'MaxDataDiskCount=64', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']  None
virtualMachines  westeurope   Standard_DS14_v2    1,2,3    ['MaxResourceVolumeMB=229376', 'OSVhdSizeMB=1047552', 'vCPUs=16', 'MemoryGB=112', 'MaxDataDiskCount=64', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']  None
virtualMachines  westeurope   Standard_DS15_v2    1,2,3    ['MaxResourceVolumeMB=286720', 'OSVhdSizeMB=1047552', 'vCPUs=20', 'MemoryGB=140', 'MaxDataDiskCount=64', 'LowPriorityCapable=True', 'PremiumIO=True', 'EphemeralOSDiskSupported=True']  None
```
For more information see: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-sku-not-available-errors

I'm not sure if this affects also the machine-controller or Kubermatic API itself, so feel free to create more Issues based on this one. 

In generell we should implement this for all cloud provider, where it is possible, so that the user get less chance to do a failure. A quick win could also be to add info message, where a end user can find the information how to find the right machine type.

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: filter out not valid VM types for azure provider. Checks SKUs restrictions for a given location and return only supported VM types

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2511

```release-note
filter out not valid VM types for azure provider
```

### CODE REVIEW

File: api/pkg/handler/v1/provider/azure.go [hunk 1]
```diff
@@ -17,28 +17,115 @@ import (
...
+
+func (s *azureClientSetImpl) ListVMSize(ctx context.Context, location string) ([]compute.VirtualMachineSize, error) {
+	sizesResult, err := s.vmSizeClient.List(ctx, location)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list sizes: %v", err)
+	}
+	return *sizesResult.Value, nil
+}
+
+func isTierStandard(sku compute.ResourceSku) bool {
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/provider/azure.go]: why do we want to restrict machines to be only in "standard" tier ?

Comment (zreigz) [reply api/pkg/handler/v1/provider/azure.go]: removed

**Comment by zreigz** [comment]
@p0lyn0mial PTAL

**Pull request merged**